flow/Knobs.cpp

```diff
@@ -135,6 +135,7 @@ void FlowKnobs::initialize(bool randomize, bool isSimulated) {
 	init( TSC_YIELD_TIME,                                  1000000 );
 	init( MIN_LOGGED_PRIORITY_BUSY_FRACTION,                  0.05 );
 	init( CERT_FILE_MAX_SIZE,                      5 * 1024 * 1024 );
+	init( TLS_HANDSHAKE_THREADS,                                10 );
```

Comment:
Do we set the size of the stacks for these threads anywhere?  We could have more threads with smaller stacks.

Maybe add a comment pointing out that with one fdbserver process per core, this will let us have 10 threads per core, which seems like plenty (since this mechanism exists to handle thundering herd problems).

Reply:
Looks like we don't set the stack space currently.  To add it properly would mean adding it to `IThreadPool::addThread()` and its implementations, and in the underlying platform specific `create_thread()`.  It seems like a good idea, I'll try to at least get it working on systems with pthread support.

Reply:
Since the background threads are waiting synchronously on network IO, I hesitate to set an expectation of how many of these threads can consume a single CPU core or how many would be needed typically.  For higher latency connections it could take hundreds of threads while only using a couple of CPU cores.

With the stack sizes set small, it should be cheap enough to have a large number of threads so that the other TLS limit knobs are basically in control of how much handshaking is done.

Reply:
Added a stack size option, default is 64k.

Unfortunately, the threads seem to allocate an additional 65MB each, after their first handshake.  Allocated memory is stable after that, and resident memory is around 200k per thread and stable.

I'm not sure if fdbserver's memory limit enforcement is operating on allocated or resident memory, but we should probably try to figure out what this memory allocation is and prevent it if possible.